model_analyzer/config/generate/manual_model_config_generator.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from .base_model_config_generator import BaseModelConfigGenerator
+from .generator_utils import GeneratorUtils
+
+from model_analyzer.triton.model.model_config import ModelConfig
+
+from model_analyzer.model_analyzer_exceptions \
+    import TritonModelAnalyzerException
+
+
+class ManualModelConfigGenerator(BaseModelConfigGenerator):
+    """ Given a model, generates model configs in manual search mode """
+
+    def __init__(self, config, model, client):
+        """
+        Parameters
+        ----------
+        config: ModelAnalyzerConfig
+        model: The model to generate ModelConfigs for
+        client: TritonClient
+        """
+        super().__init__(config, model, client)
+
+        self._remote_mode = config.triton_launch_mode == 'remote'
+        self._reload_model_disable = config.reload_model_disable
+        self._num_retries = config.client_max_retries
+        self._search_disabled = config.run_config_search_disable
+        self._curr_config_index = 0
+        self._configs = self._generate_model_configs()
+
+    def _done_walking(self):
+        return len(self._configs) == self._curr_config_index + 1
+
+    def _step(self):
+        self._curr_config_index += 1
+
+    def _get_next_model_config(self):
+        return self._configs[self._curr_config_index]
+
+    def _generate_model_configs(self):
+        """ Generate all model config combinations """
+        if self._remote_mode:
+            configs = self._generate_remote_mode_model_configs()
+        else:
+            configs = self._generate_direct_modes_model_configs()
+
+        self._finalize_configs(configs)
+
+        return configs
+
+    def _generate_remote_mode_model_configs(self):
```

Comment:
Why isn't this also part of the base class?

Reply:
I moved _make_remote_model_config to base, as we discussed